### ISSUE TITLE
Cannot start on Wayland

### ISSUE THREAD

**Issue by filipwiech** [zed-industries/zed#13238]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed fails to start on Wayland with the following error message:
`./zed-editor: symbol lookup error: /lib/x86_64-linux-gnu/libVkLayer_MESA_device_select.so: undefined symbol: wl_proxy_marshal_flags`
The previous nightly build from yesterday worked fine, so something must have changed very recently.

### Environment

Zed: v0.141.0 (Zed Nightly 9af4b6bfc757b09f30e4cc48c1c7158dd590606a)
OS: Linux Wayland ubuntu 24.04 (GNOME)
Memory: 31 GiB
Architecture: x86_64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by filipwiech** [comment]
To clarify:
When staring Zed through the `bin/zed` as usual, nothing happens, there's no window or any message and the logs always end with the following lines:
```
2024-06-18T23:49:22+02:00 [INFO] Enabling Vulkan Portability
2024-06-18T23:49:22+02:00 [INFO] Enabling color space support
```

The error from the first comment shows in the terminal when trying to run from `libexec/zed-editor` (there's nothing in the logs). Both ways work just fine in the X11 mode (with the `WAYLAND_DISPLAY=`). :+1:

**Comment by filipwiech** [comment]
Seems that this has been fixed in the latest nightly build by #13241, thanks to @mikayla-maki and @ConradIrwin for a quick solution. :slightly_smiling_face: Closing the issue. :+1:

**Comment by redthor** [comment]
I've just installed Ubuntu 24.10 and come across this. `WAYLAND= zed` stops zed from crashing.
